### DISCUSSION TITLE
Json with functions

### DISCUSSION THREAD

**Discussion by cambo2015** [appwrite/appwrite#1247] [Q&A] [answered]
Thanks again for help. I know I ask a lot of questions and I am grateful. 
     I have spent some time researching how to send json to a function and access it inside that same function, but haven't found any answers. The second thing is how do I  get the data back. I have tried createExecution(funcId) but even when I console.log, the stdout json is undefined. I am using appwrite 0.8 and nodejs.

**Comment by cambo2015** [comment]
It looks like you can access the data in the function by using an env variable. APPWRITE_FUNCTION_DATA. I got the answer from discord. 

console.log("Hi" + process.env.APPWRITE_FUNCTION_DATA).

Still not sure how to send the data though.

**Comment by cambo2015** [comment, marked as answer]
You can send data when you call the create execution fn. It is the second parameter, but will have to parse it as a string. JSON.stringify(json)

Example: 

let promise = sdk.functions.createExecution('[FUNCTION_ID]',data);

promise.then(function (response) {
    console.log(response); // Success
}, function (error) {
    console.log(error); // Failure
});

**Reply by priyarishi** [reply]
Please provide the complete function defination..the format of data. Not working for me.

**Comment by kodumbeats** [comment]
Glad we were able to help on Discord :)

**Reply by cambo2015** [reply]
Thanks again 🎉

**Comment by Mehdi-YC** [comment]
please and what about getting json result from the function ? instead of std out , return data, i have an AI model that returns some data to be used in the front end , how can i do it
